core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -209,6 +212,18 @@ impl SchedulerController {
                 continue;
             };
 
+            // Check transaction does not have too many or duplicate locks.
+            // If it does, transaction is not valid and should be dropped here.
+            if SanitizedTransaction::validate_account_locks(
+                transaction.message(),
+                transaction_account_lock_limit,
+            )
+            .is_err()
+            {
+                saturating_add_assign!(self.count_metrics.num_dropped_on_validate_locks, 1);
+                continue;
+            }
```

Comment:
perhaps move this all the way up to `ImmutableDeserializedPacket::new()` or somewhere during packet receiving to drop bad packets before inserting into storage?

Reply:
This is done before we insert to storage.
We can't do it in `ImmutableDeserializedPacket::new` because we haven't resolved look up tables at that point.
This is immediately after we resolve look up tables, which I think is the earliest we can do this.

Potentially, it could/**should** be done in `SanitizedTransaction::try_new` but that's changing the definition of a `SanitizedTransaction`. Not sure we want to do that?

Reply:
I got confused by thread-local-MI and central-scheduler code. So this is somewhat parallel to `unprocessed_transactino_storage::consume_scan_should_process_packet()` where immmutable_deserialized_packet is sanitized then accounts locks checked, which should be OK then.

Leave `SanitizedTransaction` to solana-transaction-types crate :D, there maybe can add a type for "sanitized and account locks validated" type